src/main/java/uk/gov/hmcts/divorce/caseworker/service/ReIssueApplicationService.java

```diff
@@ -55,12 +57,16 @@
     public CaseDetails<CaseData, State> process(final CaseDetails<CaseData, State> caseDetails) {
         ReissueOption reissueOption = caseDetails.getData().getApplication().getReissueOption();
 
-        log.info("For case id {} reissue option selected is {} ", caseDetails.getId(), reissueOption.getLabel());
-
-        caseDetails.getData().getApplication().setReissueDate(now(clock));
+        log.info("For case id {} reissue option selected is {} ", caseDetails.getId(), reissueOption);
 
         var updatedCaseDetails = updateCase(caseDetails, reissueOption);
 
+        if (isNull(updatedCaseDetails)) {
+            throw new ReissueProcessingException(
```

Comment:
It might be cleaner to make updateCase impossible to return a null, and instead add this exception as the default for the switch statement

Reply:
updated code as discussed